components/admin_console/highlight.tsx

```diff
@@ -6,20 +6,23 @@ import PropTypes from 'prop-types';
 import Mark from 'mark.js';
 import debounce from 'lodash/debounce';
 
-export default class Highlight extends React.Component {
-    static propTypes = {
-        filter: PropTypes.string.isRequired,
-        children: PropTypes.node.isRequired,
-    }
+type Props = {
+    filter : string;
+    children : React.ReactNode;
+}
+
+export default class Highlight extends React.Component<Props> {
+    private markInstance? : Mark;
+    private ref : React.RefObject<HTMLDivElement>;
 
-    constructor(props) {
+    public constructor(props : Props) {
         super(props);
-        this.markInstance = null;
-        this.ref = React.createRef();
+        this.markInstance = undefined;
```

Comment:
I think this line is not needed, by default an uninitialized attribute should be undefined.

Reply:
Thanks! Fixed on commit 1d2ecb1.